src/components/Pages/Dashboard/Progress/Progress.tsx

```diff
@@ -0,0 +1,44 @@
+import { Box } from "@mui/material";
+import Streak from "./Streak/Streak";
+import HabitStrength from "./HabitStrength/HabitStrength";
+import { useSelector } from "react-redux";
+import { RootState } from "../../../../app/store";
+import { HabitData } from "../../../../app/features/habit/habitSlice";
+
+const getStreakFromData = (data: HabitData[]) => {
```

Comment:
Maybe just `calculateStreak`? the `fromData` is very generic and a bit useless IMO.

Reply:
Or `calculateStreakFromHabitData`